src/data-model-providers/codegen/EmberAttributeStorageImpl.cpp

```diff
@@ -0,0 +1,164 @@
+/**
+ *    Copyright (c) 2025 Project CHIP Authors
+ *
+ *    Licensed under the Apache License, Version 2.0 (the "License");
+ *    you may not use this file except in compliance with the License.
+ *    You may obtain a copy of the License at
+ *
+ *        http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *    Unless required by applicable law or agreed to in writing, software
+ *    distributed under the License is distributed on an "AS IS" BASIS,
+ *    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *    See the License for the specific language governing permissions and
+ *    limitations under the License.
+ */
+#include "lib/core/CHIPError.h"
+#include <data-model-providers/codegen/EmberAttributeStorageImpl.h>
+
+#include <app-common/zap-generated/attribute-type.h>
+#include <app/util/persistence/AttributePersistenceProvider.h>
+#include <lib/support/ScopedBuffer.h>
+
+namespace chip {
+namespace app {
+namespace Storage {
+namespace {
+
+// EmberAfAttributeType possible values are part of the unnamed enum in `attribute-type.h`,
+// of the form ZCL_*_ATTRIBUTE_TYPE
+EmberAfAttributeType attributeType(AttributeStorage::Buffer & buffer)
+{
+    switch (buffer.type())
+    {
+    case AttributeStorage::Buffer::Type::kPrimitive:
+        switch (buffer.size())
+        {
+        case 1:
+            return ZCL_INT8U_ATTRIBUTE_TYPE;
```

Comment:
Please document why it's ok to assume unsigned here.

More generally, see the comments below about how we are making assumption about AttributePersistenceProvider implementations that need to be documented in that API...  Here we are assuming that the only thing it needs out of the type is whether it's a short/long string of some sort or not.